pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -0,0 +1,260 @@
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		reqLogger.Error(err, "error looking up clusterDeployment")
+		return reconcile.Result{}, err
+	}
+
+	if err := r.syncCertificateRequests(cd, reqLogger); err != nil {
+		reqLogger.Error(err, "error syncing CertificateRequests")
+		return reconcile.Result{}, err
+	}
+
+	reqLogger.Info("done syncing")
+	return reconcile.Result{}, nil
+}
+
+func (r *ReconcileClusterDeployment) syncCertificateRequests(cd *hivev1alpha1.ClusterDeployment, logger logr.Logger) error {
+	desiredCRs := []certmanv1alpha1.CertificateRequest{}
+
+	// get a list of current CertificateRequests
+	currentCRs, err := r.getCurrentCertificateRequests(cd, logger)
+	if err != nil {
+		return err
+	}
+
+	// for each certbundle with generate==true make a CertificateRequest
+	for _, cb := range cd.Spec.CertificateBundles {
+		if cb.Generate == true {
+			domains := getDomainsForCertBundle(cb, cd)
+
+			certReq := createCertificateRequest(cb.Name, domains, cd)
+			desiredCRs = append(desiredCRs, certReq)
+		}
+	}
+
+	deleteCRs := []certmanv1alpha1.CertificateRequest{}
+
+	// find any extra certificateRequests and mark them for deletion
+	for i, currentCR := range currentCRs {
+		found := false
+		for _, desiredCR := range desiredCRs {
+			if desiredCR.Name == currentCR.Name {
+				found = true
+				break
+			}
+		}
+		if !found {
+			deleteCRs = append(deleteCRs, currentCRs[i])
+		}
+	}
+
+	// create/update the desired certificaterequests
+	for _, desiredCR := range desiredCRs {
+		currentCR := &certmanv1alpha1.CertificateRequest{}
+		searchKey := types.NamespacedName{Name: desiredCR.Name, Namespace: desiredCR.Namespace}
+
+		if err := r.client.Get(context.TODO(), searchKey, currentCR); err != nil {
+			if errors.IsNotFound(err) {
+				// create
+				if err := controllerutil.SetControllerReference(cd, &desiredCR, r.scheme); err != nil {
+					logger.Error(err, "error setting owner reference", "certrequest", desiredCR.Name)
+					return err
+				}
+
+				if err := r.client.Create(context.TODO(), &desiredCR); err != nil {
+					logger.Error(err, "error creating certificaterequest")
+					return err
+				}
+			} else {
+				logger.Error(err, "error checking for existing certificaterequest")
+				return err
+			}
+		} else {
+			// update or no updated needed
+			if !reflect.DeepEqual(currentCR.Spec, desiredCR.Spec) {
+				if err := r.client.Update(context.TODO(), currentCR); err != nil {
+					logger.Error(err, "error updating certificaterequest", "certrequest", currentCR.Name)
+					return err
+				}
+			} else {
+				logger.Info("no updated needed for certificaterequest", "certrequest", desiredCR.Name)
```

Comment:
s/updated/update/

Reply:
You also likely meant to have logger.WithField("certrequest", desiredCR.Name).Info(...)

Reply:
@csrwng i'm just following along with what the default logger library is for operator-sdk (go-logr)...it takes the key=value pairs without having to do the .WithField(): https://github.com/openshift/certman-operator/blob/master/vendor/github.com/go-logr/logr/logr.go#L111

Reply:
sorry @joelddiaz I simply assumed you were using the sirupsen logger, please ignore me :)